server/xcode/xcode_darwin.go

```diff
@@ -129,13 +141,34 @@ func versionMap(urlRefs []C.CFURLRef) map[string]*xcodeVersion {
 				continue
 			}
 			log.Debugf("Mapped Xcode Version %s=>%s with SDKs %+v", version, developerDirPath, sdks)
-			versionMap[version] = &xcodeVersion{
+			xv := &xcodeVersion{
 				version:          mostPreciseVersion,
 				developerDirPath: developerDirPath,
 				sdks:             sdks,
 			}
+			versionMap[version] = xv
+			if defaultDeveloperDir == developerDirPath {
+				versionMap[defaultXcodeVersion] = xv
+			}
+		}
+	}
+
+	if xv, ok := versionMap[defaultXcodeVersion]; ok {
+		log.Infof("xcode-select set to Xcode version %s", xv.version)
+		return versionMap
+	}
+
+	// If we are here then defaultDeveloperDir is not a valid Xcode path
+	if defaultDeveloperDir == "/Library/Developer/CommandLineTools" {
+		version := commandLineToolsVersionString()
+		log.Infof("xcode-select set to Command Line Tools (%s)", version)
+		if xv := commandLineToolsVersion(defaultDeveloperDir); xv != nil {
+			versionMap[defaultXcodeVersion] = xv
 		}
+	} else if defaultDeveloperDir != "" {
+		log.Warningf("xcode-select set to an invalid Xcode path: %s", defaultDeveloperDir)
```

Comment:
it's a little weird to have this else case here just to detect the special case when defaultDeveloperDir == "" and then log the error that xcode-select was not found. Could we log this warning inside of xcodeSelectDeveloperDir and then omit this else if block?

Reply:
When `defaultDeveloperDir != ""`, which will only happen when `xcode-select` runs correctly, but it's set to a no longer installed Xcode, or some other invalid Xcode path. We only know it's invalid at this point, instead of in `xcodeSelectDeveloperDir`, after we've tried to match the path to all installed Xcode versions detected.